relay-metrics/src/protocol.rs

```diff
@@ -54,3 +97,59 @@ pub(crate) fn hash_set_value(string: &str) -> u32 {
     hasher.write(string.as_bytes());
     hasher.finish32()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_unescape_tag_value() {
```

Comment:
We should should have some tests:

- for characters outside of ascii (chinese etc.)
- for other control characters, that they are replaced correctly (the function still calls into the old `validate`)